error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -78,6 +78,38 @@ Mono<T> after(Optional<T> optional) {
     }
   }
 
+  /**
+   * Try to avoid expressions of type {@code Optional<Mono<T>>}, but if you must map an {@link
+   * Optional} to this type, prefer using {@link Mono#just(Object)}.
+   */
+  static final class OptionalMapMonoJust<T> {
+    @BeforeTemplate
+    Optional<Mono<T>> before(Optional<T> optional) {
+      return optional.map(Mono::justOrEmpty);
+    }
+
+    @AfterTemplate
+    Optional<Mono<T>> after(Optional<T> optional) {
+      return optional.map(Mono::just);
+    }
+  }
+
+  /**
+   * Prefer a {@link Mono#justOrEmpty(Optional)} and {@link Mono#switchIfEmpty(Mono)} chain over
```

Comment:
I know this is our usual way of defining the Javadoc. 
However, it couldn't hurt to mention the exact reason either as `@Description` or instead of `over more contrived alternatives`. Not mixing the `Optional` and `Mono` API is a good argument and nice to document here as well.

We could IMO at least start with adding `@Description`s here and there.

Reply:
I can try to come up with something later today. :crossed_fingers:

Reply:
Let's go with Javadoc then, deal?

Reply:
Rebased and added a commit; PTAL.

Reply:
LGTM!

Reply:
Yeah, would indeed feel like a different concern. Would have been weird starting this now in this PR. :thinking: